react/components/SliderContext.tsx

```diff
@@ -206,19 +207,49 @@ const SliderContextProvider: FC<SliderContextProps> = ({
 
   const newSlides = preRenderedSlides.concat(slides, postRenderedSlides)
 
-  const slideWidth = useMemo(() => 100 / newSlides.length, [newSlides.length])
+  const slideWidth = useMemo(() => {
+    const baseSlideWidth = 100 / newSlides.length
+
+    let resultingSlideWidth = baseSlideWidth
+
+    if (centerMode !== 'disabled') {
+      resultingSlideWidth =
+        (resolvedSlidesPerPage / (resolvedSlidesPerPage + 1)) * baseSlideWidth
```

Comment:
![](https://media.giphy.com/media/HBWbIuHvXI2Eo/giphy.gif)

Reply:
I saw this one yesterday!! Laughed a lot 😂